src/modules/expression/binop/range.rs

```diff
@@ -59,17 +59,60 @@ impl SyntaxModule<ParserMetadata> for Range {
 impl TranslateModule for Range {
     fn translate(&self, meta: &mut TranslateMetadata) -> String {
         let from = self.from.translate(meta);
-        let to = self.to.translate(meta);
-        if self.neq {
-            let to_neq = if let Some(ExprType::Number(_)) = &self.to.value {
-                to.parse::<isize>().unwrap_or_default().sub(1).to_string()
+        let to = if let Some(to) = self.to.get_integer_value() {
+            if self.neq {
+                (to - 1).to_string()
             } else {
-                translate_computation(meta, ArithOp::Sub, Some(to), Some("1".to_string()))
-            };
-            meta.gen_subprocess(&format!("seq {} {}", from, to_neq))
+                to.to_string()
+            }
         } else {
-            meta.gen_subprocess(&format!("seq {} {}", from, to))
+            let to = self.to.translate(meta);
+            if self.neq {
+                translate_computation(meta, ArithOp::Sub, Some(to), Some("1".to_string()))
+            } else {
+                to
+            }
+        };
+        let stmt = format!("seq {} {}", from, to);
+        meta.gen_subprocess(&stmt)
+    }
+}
+
+impl Range {
+    pub fn get_array_index(&self, meta: &mut TranslateMetadata) -> (String, String) {
+        if let Some(from) = self.from.get_integer_value() {
```

Comment:
Use compile-time arithmetic to convert Amber `array[10..=12]` to Bash `${array[@]:10:3}`; fall back to calling `bc` if not supplied a number.